pkg/serial/handshake.go

```diff
@@ -54,206 +59,208 @@ func PurgeIncoming(conn net.Conn) {
 	conn.SetReadDeadline(time.Time{})
 }
 
-func nextPos(syncPos byte) byte { return (syncPos + 1) | 0x80 }
+func incrementByte(syncPos byte) byte { return (syncPos + 1) | 0x80 }
 
-func HandshakeClient(ctx context.Context, conn net.Conn, debug bool) error {
+// HandshakeClient establishes connection with the server making sure
+// they both are in sync.
+func HandshakeClient(conn io.ReadWriter, debug bool) error {
 	if tracing {
 		defer trace.End(trace.Begin(""))
 	}
 
 	buf1byte := make([]byte, 1)
-	rand.Read(buf1byte)
-	pos := nextPos(buf1byte[0])
+
+	if _, err := rand.Read(buf1byte); err != nil {
+		log.Errorf("HandshakeClient: Could not read a random byte due to: %v", err)
+	}
+
+	pos := incrementByte(buf1byte[0])
 
 	// set the read deadline for timeout
 	// this has no effect on windows as the deadline is set at port open time
-	deadline, ok := ctx.Deadline()
-	if ok {
-		conn.SetReadDeadline(deadline)
-	}
 
-	log.Debugf("HandshakeClient: Establishing connection...")
-	for {
-		log.Debug("HandshakeClient: Sending syn.")
-		conn.Write([]byte{flagSyn, pos})
-		pos = nextPos(pos)
+	log.Debug("HandshakeClient: Sending syn.")
```

Comment:
This is a negligible amount of logging comparable to what is logged before, it is even hard to find a trace of it.